mlflow_elasticsearchstore/elasticsearch_store.py

```diff
@@ -114,15 +115,22 @@ def _list_experiments_name(self) -> List[str]:
         response = s.execute()
         return [name.key for name in response.aggregations.exp_names.buckets]
 
+    def _get_artifact_location(self, experiment_id: str) -> str:
+        return append_to_uri_path(self.artifact_root_uri, str(experiment_id))
+
     def create_experiment(self, name: str, artifact_location: str = None) -> str:
         if name is None or name == '':
             raise MlflowException('Invalid experiment name', INVALID_PARAMETER_VALUE)
         existing_names = self._list_experiments_name()
         if name in existing_names:
             raise MlflowException('This experiment name already exists', INVALID_PARAMETER_VALUE)
+
         experiment = ElasticExperiment(name=name, lifecycle_stage=LifecycleStage.ACTIVE,
                                        artifact_location=artifact_location)
         experiment.save(refresh=True)
+        if not artifact_location:
+            artifact_location = self._get_artifact_location(experiment.meta.id)
+        experiment.update(refresh=True, artifact_location=artifact_location)
```

Comment:
Why do you check and update after inserting the experiment ? You could check the value of artifact location before the save and get the default one if it is none

Reply:
I need the experiment_id which is assigned during save

Reply:
If I want to do it before, either I do like create_run and I create the before id or I use something else which is unique like experiment_name for example